SUBSCRIPTION.md

```diff
@@ -0,0 +1,442 @@
+# Virtual Labs Subscription System
+
+## Overview
+
+The Virtual Labs platform offers a tiered subscription model designed to accommodate different user needs and usage patterns. This document outlines the subscription system's architecture, functionality, and workflows.
+
+```mermaid
+graph TD
+    A[User Creates Virtual Lab] --> B[Free Subscription Activated]
+    B --> C{Choose Subscription}
+    C -->|Upgrade| D[Pro/Premium Plan]
+    C -->|Stay on| E[Free Plan]
+    D --> F[Access Advanced Features]
+    D --> G[Receive Monthly/Yearly Credits]
+    E --> H[Limited Features]
+    E --> I[Basic Credits]
+    D -->|Top-up| J[Add Additional Credits]
+    E -->|Top-up| J
+    D -->|Payment Fails| E
```

Comment:
The graph at the moment doesn't cover the case where a subscription is made before creating a virtual lab.

Also, would be nice to add a "start" node, so that the initial state is clear.